base/array.jl

```diff
@@ -740,7 +740,7 @@ findfirst(A) = findnext(A, 1)
 # returns the index of the next matching element
 function findnext(A, v, start::Integer)
     for i = start:length(A)
-        if A[i] == v
+        if isequal(A[i], v)
```

Comment:
this should be documented and tested as a behavior change on findnext/findprev with -0.0